### PR TITLE
Intel: remove Intel XTOS platform code and drivers

### PR DESCRIPTION
**Pull request by kv2019i**
To complete the transition to Zephyr for Intel hardware targets, remove the bulk of platform code and drivers only used by Intel cAVS2.5 target in the non-Zephyr XTOS build. This is no longer supported (and hasn't been tested at all in recent times), so time to remove this.

Note: there is still some remaining bits in the codetree that will be removed in follow-up PRs. The platform driver headers need a bit more work to remove (due to depencies from rtos headers), the cAVS2.5 LPS code and a few other bits.

### CODE REVIEW

File: zephyr/CMakeLists.txt [hunk 1]
```diff
@@ -118,81 +118,22 @@ zephyr_include_directories(
...
-		${SOF_PLATFORM_PATH}/intel/cavs/lib/pm_memory.c
-		${SOF_PLATFORM_PATH}/intel/cavs/lib/clk.c
-		${SOF_PLATFORM_PATH}/intel/cavs/lib/dma.c
-	)
-	set_source_files_properties(${SOF_PLATFORM_PATH}/intel/cavs/lib/power_down.S PROPERTIES COMPILE_FLAGS -DASSEMBLY)
-	else()
-	zephyr_library_sources(
-		${SOF_PLATFORM_PATH}/intel/cavs/platform.c
-		${SOF_PLATFORM_PATH}/intel/cavs/lib/mem_window.c
 		${SOF_PLATFORM_PATH}/tigerlake/lib/clk.c
```
Comment (btian1) [inline zephyr/CMakeLists.txt:132]: mem_windows.c got removed as well?

**Review by RanderWang — APPROVED** [review]
LGTM

File: zephyr/CMakeLists.txt (hunk 1 above)
Comment (ujfalusi) [reply zephyr/CMakeLists.txt:132]: mem windows are handled by Zephyr, we have independent PRs to disable/remove it.

**Review by ujfalusi — APPROVED** [review]
I was not brave enough to take on this, thanks!

**Comment by kv2019i** [comment]
Hmm. Intel internal test "fw_113_ssp/test_02_ssp_internal_loop.py::TestSspInternalLoop::test_113_02_simple_internal_loopback" fails, but seems unlikely to be related to this PR (platform this PR does not touch). Will need a rerun to check.

**Review by singalsu — APPROVED** [review]
Approve for DMIC, thanks Kai.

File: zephyr/CMakeLists.txt (hunk 1 above)
Comment (kv2019i) [reply zephyr/CMakeLists.txt:132]: Split into a separate patch, resolving.

**Comment by kv2019i** [comment]
@wszypelt Can you check this? This is failing the same TestSspInternalLoop.test_113_02_simple_internal_loopback test again, but I simply can't see a connection to the patch. Any insight, if same test case passing on other runs?

**Review by lgirdwood — APPROVED** [review]

**Review by btian1 — COMMENTED** [review]
like remove things, :)

File: zephyr/CMakeLists.txt (hunk 1 above)
Comment (ujfalusi) [inline zephyr/CMakeLists.txt:132]: Should you also remove the tigerlake/lib/clk.c from the zephyr build? You remove the file...

Comment (kv2019i) [reply zephyr/CMakeLists.txt:132]: @ujfalusi  I don't remove platfrom/tigerlake/lib/clk.c , it's still there. I do remove platform/intel/cavs/lib/clk.c , but that is not referred to by Zephyr cmake rules (for cAVS2.5).

Comment (ujfalusi) [reply zephyr/CMakeLists.txt:132]: oh, true. Too many, too different things.

**Comment by lgirdwood** [comment]
@wszypelt @lrudyX unused code removal so not expecting a failure. Good to merge ?

**Comment by kv2019i** [comment]
V2: 
- found a weird (but real) issue in SSP driver that explains the fail -- will need to follow-up in Zephyr upstream to address this and then we can later remove the TODO entry in this series (but this is good for now)

**Comment by kv2019i** [comment]
Yiihaa, "Internal CI System" now passes, so the fix helped. Need to wait until Monday to get the SOF CI results.

**Comment by kv2019i** [comment]
Proceeding with merge. Two known fails in https://sof-ci.01.org/sofpr/PR8048/build11827/devicetest/index.html

**Pull request merged**